components/layout/display_list/stacking_context.rs

```diff
@@ -992,8 +994,11 @@ impl BoxFragment {
             return;
         }
 
+        let frame_origin_for_query = self.cumulative_border_box_rect().origin.to_webrender();
+
         let new_spatial_id = stacking_context_tree.push_reference_frame(
             reference_frame_data.origin.to_webrender(),
+            frame_origin_for_query,
```

Comment:
Is this truly the right value to use for this? It isn't taking into account any of the information from ancestors such as scroll offsets and transforms...

Reply:
Yes, we need this find the correct offset (relative to the document) when a transformation happens. And, the offset of the given rect in the queries was without any transform (and in fact, is using the cumulative box as well).